### DISCUSSION TITLE
Is there a way to require() markdown files without having them pre-processed as MDX components?

### DISCUSSION THREAD

**Discussion by ZenaMel** [facebook/docusaurus#8731] [Q&A] [answered]
Scenario: I'm writing a custom component that turns regular markdown tables into filterable tables (with search functionality). Currently, I pass JSON table files or Markdown table files (.md files under /static/tables/ that contain a single table, nothing else).

Getting the JSON tables is no problem with

```js
        if (fileSource.endsWith('.json')) {
            const json = require(`@site/static/tables/${tableSourceFileName}`)
            const data = json2FilterTableData(json)
            columns = data.columns
            rows = data.rows
        }   
```

when fetching a markdown table file via the same `require()` syntax, I get some type of transformed MDX component back instead of the raw file contents.

Is there a way around this?

What I already tried:

- renaming the table files from `.md` to `.md_` so they don't get preprocessed by Docusaurus. I couldn't get this to work with webpack, although I specified a custom loader for this.
- using `.txt` as an extension. Same problem as above.

**Comment by slorber** [comment, marked as answer]
https://docusaurus.io/docs/markdown-features/react#importing-code-snippets

MD/MDX is treated as code by Docusaurus thanks to Webpack loaders, but it's possible to opt-out of this and get a string instead:

```js
import MyComponentString from '!!raw-loader!./myComponent';

import MyMarkdownString from '!!raw-loader!./myMarkdownFile.md';
```

This is a Webpack proprietary feature so we don't particularly recommend it as we might want to migrate away from Webpack someday this might brake or the API to enable this might change.

**Reply by ZenaMel** [reply]
Thanks! I'll keep that in mind.

**Comment by ZenaMel** [comment]
@slorber we're using Webpack 5.x since at least 2.3.1 right? Looks like this method is recommended now: https://webpack.js.org/guides/asset-modules/

I wasn't able to get your suggestion working as the raw loader would try and parse the file as JSON. Maybe I need to configure it correctly? ...

**Reply by slorber** [reply]
Stale PR trying to use this new system: https://github.com/facebook/docusaurus/pull/4708

I don't know why it doesn't work in your case or for JSON. It's a proprietary bundler feature so you need to read the Webpack doc and figure it out. We only documented it lightly to show it's possible to achieve, but do not have the knowledge of all the possible edge cases.
